mycpp/marksweep_heap.cc

```diff
@@ -0,0 +1,114 @@
+#include "mycpp/runtime.h"
+
+void MarkSweepHeap::Init(int max_heap_bytes) {
+  this->max_heap_bytes = max_heap_bytes;
+}
+
+void* MarkSweepHeap::Allocate(int byte_count) {
+  void* Result = calloc(byte_count, 1);
+  assert(Result);
+
+  this->AllAllocations.insert(Result);
+
+  this->current_heap_bytes += byte_count;
+  if (this->current_heap_bytes > this->max_heap_bytes) {
+    Collect();
+  }
+
+  if (this->current_heap_bytes > this->max_heap_bytes) {
+    // NOTE(Jesse): Generally, doubling results in a lot of wasted space.  I've
+    // observed growing by a factor of 1.5x, or even 1.3x, to be a good
+    // time-space tradeoff.
+    //
+    // 1.5x = (3/2)
+    // 1.3x = (13/10)
+    //
+    this->max_heap_bytes = this->current_heap_bytes * 3 / 2;
+  }
+
+  return Result;
+}
+
+void MarkSweepHeap::MarkAllReferences(Obj* obj) {
+  auto header = ObjHeader(obj);
+
+  this->MarkedAllocations.insert((void*)obj);
```

Comment:
Please use C++ style casts, e.g. static_cast or reinterpret_cast 

I think that is consistent everywhere in the codebase

Reply:
Fixed